test/dev/dev.test.ts

```diff
@@ -0,0 +1,59 @@
+const path = require('path');
+
+const execa = require('execa');
+const {readdirSync, readFileSync, statSync, existsSync} = require('fs');
+const glob = require('glob');
+const os = require('os');
+const got = require('got');
+
+describe('snowpack dev', () => {
+  let snowpackProcess;
+  afterEach(async () => {
+    snowpackProcess.cancel();
+
+    try {
+      await snowpackProcess;
+    } catch (error) {
+      expect(error.isCanceled).toEqual(true);
+    }
+  });
+
+  it('smoke', async () => {
+    expect.assertions(3);
+
+    const cwd = path.join(__dirname, 'smoke');
+
+    // start the server
+    // NOTE: we tried spawning `yarn` here, but the process was not cleaned up
+    //       correctly on CI and the action got stuck. npx does not cause that problem.
+    snowpackProcess = execa('npx', ['snowpack', 'dev'], {cwd});
```

Comment:
I tried to spawn `yarn`, but it got stuck. I tried using `snowpackProcess.kill()` with the `forceKillAfterTimeout` option, but it still got stuck. That means we can't just define a `"teststart"` script in each `<folder>/package.json`, which is not ideal. 

I've been debugging this for quite a while, happy to test other approaches, but maybe we should merge this and try to fix it in a follow up PR

Reply:
- I'd be curious if `windowsHide: false` or `shell: true` do anything for us here.
- piping snowpackProcess .stdout & snowpackProcess.stderr to the console would also be interesting.